src/lib.rs

```diff
@@ -45,7 +45,17 @@ pub fn depth(i: usize) -> usize {
 }
 
 /// Returns the offset of a node with a depth.
+///
+/// ## Examples
+/// ```rust
+/// assert_eq!(flat_tree::offset_with_depth(0, 0), 0);
+/// assert_eq!(flat_tree::offset_with_depth(1, 1), 0);
+/// assert_eq!(flat_tree::offset_with_depth(2, 0), 1);
+/// assert_eq!(flat_tree::offset_with_depth(3, 2), 0);
+/// assert_eq!(flat_tree::offset_with_depth(4, 0), 2);
+/// ```
 pub fn offset_with_depth(i: usize, depth: usize) -> usize {
+  debug_assert_eq!(depth, self::depth(i), "Invalid depth!");
```

Comment:
```suggestion
  debug_assert_eq!(depth, self::depth(i), "Invalid depth.");
```

Using an exclamation mark might come off as if we're shouting at people, instead of excitement (which I think was the intended tone here?). Instead using a period might allow us to keep a more neutral voice, which is probably what we want.